rest-api-spec/src/main/resources/rest-api-spec/test/cluster.health/10_basic.yml

```diff
@@ -92,6 +92,37 @@
   - match:     { unassigned_shards:       0 }
   - gte:       { number_of_pending_tasks: 0 }
 
+---
+"cluster health basic test, one index with wait for no initializing shards":
+  - skip:
+      version: " - 6.99.99"
+      reason: "wait_for_no_initializing_shard is introduced in 7.0.0"
+
+  - do:
+      indices.create:
+        index: test_index
+        body:
+          settings:
+            index:
+              number_of_replicas: 0
```

Comment:
creating an index automatically waits for primary to be allocated (wait_for_active_shards is 1 by default). This means that the health check later is fully redundant, it will always succeed. This test does not really test anything here.

Reply:
Removed.

Reply:
how about changing it so that the index is created with wait_for_active_shards : 0? Then this health check will actually wait for the primaries to be allocated and the output should guarantee that number of initializing shards is 0?

Reply:
Thanks @ywelsch, your suggestion really tests the new parameter :). I pushed https://github.com/elastic/elasticsearch/pull/27489/commits/04e61fbb2c0bc0436e713e17c053c0af42a20c72